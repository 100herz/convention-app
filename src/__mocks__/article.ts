import { Article } from '@models/article'

export const article: Article = {
  id: 18569,
  date_gmt: '2020-11-11T08:25:41',
  title: {
    rendered: '„New Normal“ bei Leonardo Hotels &#8211; mit neuen MICE-Formaten und Konzepten',
  },
  content: {
    rendered:
      '<p>Durch smarte Raumanpassungen, große Flexibilität bei Buchungen und Stornierungen sowie die Etablierung eines extern zertifizierten und weit über den Standard hinaus gehenden Sicherheits- und Hygienekonzeptes hat sich Leonardo Hotels umfassend auf das „New Normal“ und die geänderten Anforderungen ihrer MICE-Kunden an zukunftsfähigen Veranstaltungsformaten eingestellt. Neue Pausenkonzepte, digitalen Helfer, verlässliche Partner sowie virtuelle und hybride Veranstaltungsformate umfassen die Angebotsvielfalt der Leonardo Hotels.  Diese sind überzeugt, dass persönliche Meetings mit Sicherheit wieder Zukunft haben.</p>\n',
    protected: false,
  },
  excerpt: {
    rendered:
      '<p>Durch neue Sicherheits- und Hygienekonzeptes hat sich Leonardo Hotels umfassend auf das „New Normal“ und die geänderten Anforderungen an zukunftsfähigen Veranstaltungsformaten eingestellt.</p>\n',
    protected: false,
  },
  featured_media: 18571,
  categories: [6, 1],
  acf: {
    sponsored_by: 'XYZ',
    should_push: false,
    pinned: false,
    featured_slider: false,
    news_slider: true,
  },
  featured_image_thumb:
    'https://www.convention-net.de/wp-content/uploads/2020/11/iStock-1250329916-Personen_mit_Masken-w1500-h1300-300x300.jpg',
  featured_image_medium:
    'https://www.convention-net.de/wp-content/uploads/2020/11/iStock-1250329916-Personen_mit_Masken-w1500-h1300-600x400.jpg',
  categories_names: ['Hotels', 'News'],
}

export const articleWithoutFeatured: Article = {
  id: 18592,
  date_gmt: '2020-11-10T11:00:19',
  title: {
    rendered:
      'Events and Hospitality Industries Urged to Harness the Power of Digital Transformation to Thrive in 2021 and Beyond',
  },
  content: {
    rendered:
      '<p><a href="https://bit.ly/3iPkUhB">Cvent</a> CEO and Founder, Reggie Aggarwal, delivered an inspiring keynote to more than 7,000 event, marketing, and hospitality professionals at <a href="https://bit.ly/30Unj4A">Cvent CONNECT Europe</a> today. In his opening presentation, Aggarwal urged attendees of the virtual event to seize the digital transformation opportunity in front of them to drive the industry’s recovery in 2021 and beyond.</p>\n',
    protected: false,
  },
  excerpt: {
    rendered:
      '<p>Cvent CEO Reggie Aggarwal predicts “Golden Age of Events” during his keynote to more than 7,000 virtual attendees at Cvent CONNECT Europe </p>\n',
    protected: false,
  },
  featured_media: null,
  categories: [358, 217, 1],
  acf: {
    sponsored_by: '',
    should_push: false,
    pinned: false,
    featured_slider: false,
    news_slider: true,
  },
  featured_image_thumb: '',
  featured_image_medium: '',
  categories_names: ['Branche', 'Events', 'News'],
}

export const articleWithoutSponsored: Article = {
  id: 18543,
  date_gmt: '2020-11-09T09:46:40',
  title: {
    rendered: 'Sicheres Business und Inspiration für morgen: Koreas neue, einzigartige Veranstaltungsorte',
  },
  content: {
    rendered:
      '<figure id="attachment_18559" aria-describedby="caption-attachment-18559" style="width: 618px" class="wp-caption aligncenter"><img class="wp-image-18559 size-large" src="https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300-1200x801.jpg" alt="" width="618" height="413" srcset="https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300-1200x801.jpg 1200w, https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300-600x400.jpg 600w, https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300-768x513.jpg 768w, https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300-110x75.jpg 110w, https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300.jpg 1500w" sizes="(max-width: 618px) 100vw, 618px" /><figcaption id="caption-attachment-18559" class="wp-caption-text">Foto: Pairat Pinijkul / shutterstock.com</figcaption></figure>\n<p>Aufgrund eines exzellenten IKT-Umfelds und seiner kulturellen Attraktivität hat sich Korea als effizientes, innovatives Land für Unternehmen und Kongresse in der MICE-Branche etabliert. Hinsichtlich der Anzahl internationaler Veranstaltungen im Land belegte Korea in den letzten Jahren unter den Mitgliedern der Union der Internationalen Verbände (UIA) den ersten und zweiten Platz. Seit dem Ausbruch von Covid-19 hat die schnelle und gut organisierte Reaktion es dem Land ermöglicht, für die Sicherheit vieler Unternehmen und Konventionen zu sorgen. Mit zahlreichen Support-Programmen unterstützt Korea Verbände und Organisationen bei der Planung und Ausrichtung von Events und baut nach wie vor das Angebot an vielfältigen Eventlocations im Land aus.</p>\n',
    protected: false,
  },
  excerpt: {
    rendered:
      '<p>Aufgrund eines exzellenten IKT-Umfelds und seiner kulturellen Attraktivität hat sich Korea als effizientes, innovatives Land für Unternehmen und Kongresse in der MICE-Branche etabliert.</p>\n',
    protected: false,
  },
  featured_media: 18559,
  categories: [5, 1, 72, 170],
  acf: {
    // sponsored_by: '100HERZ',
    should_push: false,
    featured_slider: true,
    news_slider: true,
    pinned: true,
  },
  featured_image_thumb:
    'https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300-300x300.jpg',
  featured_image_medium:
    'https://www.convention-net.de/wp-content/uploads/2020/11/shutterstock_335001080-w1500-h1300-600x400.jpg',
  categories_names: ['Destinationen', 'News', 'Reportagen', 'Sponsored content'],
}
