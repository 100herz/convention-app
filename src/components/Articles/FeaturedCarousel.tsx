import React from 'react'
import { Dimensions, ImageBackground, StyleSheet, TextStyle, View, ViewStyle } from 'react-native'
import Carousel from 'react-native-snap-carousel'

import Text from '@components/UI/Text'
import { hexToRgb } from '@utils/styling'
import { Article } from '@models/article'
import { colors, defaultStyles, DefaultStyles, fonts } from '@styles/theme'

interface Props {
  articles: Article[]
}

const FeaturedCarousel: React.FC<Props> = ({ articles }) => {
  const screenWidth = Dimensions.get('screen').width

  const slideItem = ({ item }: { item: Article }) => (
    <View style={styles.container}>
      <ImageBackground
        source={{ uri: item._embedded['wp:featuredmedia'] && item._embedded['wp:featuredmedia'][0].source_url }}
        style={styles.backgroundImage}
      >
        <View
          style={{
            ...styles.slideContainer,
            justifyContent:
              item.acf.sponsored_by !== null && item.acf.sponsored_by.length > 0 ? 'space-between' : 'flex-end',
          }}
        >
          {item.acf.sponsored_by !== null && item.acf.sponsored_by.length > 0 && (
            <Text style={styles.sponsored}>
              Sponsored by <Text style={{ fontFamily: fonts.sansBold }}>{item.acf.sponsored_by}</Text>
            </Text>
          )}
          <View style={styles.titleContainer}>
            <Text style={styles.title}>{item.title.rendered}</Text>
          </View>
        </View>
      </ImageBackground>
    </View>
  )
  return (
    <Carousel
      data={articles}
      renderItem={slideItem}
      sliderWidth={screenWidth}
      itemWidth={screenWidth}
      inactiveSlideOpacity={0.5}
      inactiveSlideScale={0.8}
    />
  )
}

interface Styles extends DefaultStyles {
  slideContainer: ViewStyle
  sponsored: TextStyle
  titleContainer: ViewStyle
}

const styles = StyleSheet.create<Styles>({
  ...defaultStyles,
  container: {
    height: Dimensions.get('screen').height * 0.3,
  },
  slideContainer: {
    flex: 1,
  },
  sponsored: {
    width: 'auto',
    alignItems: 'flex-end',
    backgroundColor: `rgba(${hexToRgb(colors.grayLight)}, 0.7)`,
    paddingHorizontal: 30,
    paddingVertical: 10,
    fontSize: 14,
    textAlign: 'right',
  },
  titleContainer: {
    justifyContent: 'flex-end',
    backgroundColor: `rgba(${hexToRgb(colors.grayLight)}, 0.7)`,
    paddingHorizontal: 30,
    paddingVertical: 10,
  },
  title: {
    fontSize: 18,
  },
})

export default FeaturedCarousel
