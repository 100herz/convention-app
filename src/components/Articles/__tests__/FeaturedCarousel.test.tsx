import React from 'react'
import { render } from '@testing-library/react-native'

import FeaturedCarousel from '../FeaturedCarousel'
import { article, articleWithoutFeatured } from '@__mocks__/article'

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper')

export const mockedNavigate = jest.fn()
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native')
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
    useNavigationParam: jest.fn(jest.requireActual('@react-navigation/native').useNavigationParam),
  }
})

describe('<FeaturedCarousel />', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<FeaturedCarousel articles={[article, articleWithoutFeatured]} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
