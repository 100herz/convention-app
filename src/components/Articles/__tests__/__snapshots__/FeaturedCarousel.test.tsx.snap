// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeaturedCarousel /> renders correctly 1`] = `
<RCTScrollView
  activeAnimationOptions={null}
  activeAnimationType="timing"
  activeSlideAlignment="center"
  activeSlideOffset={20}
  apparitionDelay={0}
  automaticallyAdjustContentInsets={false}
  autoplay={false}
  autoplayDelay={1000}
  autoplayInterval={3000}
  callbackOffsetMargin={5}
  containerCustomStyle={Object {}}
  contentContainerCustomStyle={Object {}}
  contentContainerStyle={
    Array [
      Object {
        "paddingLeft": 0,
        "paddingRight": 0,
      },
      Object {},
    ]
  }
  data={
    Array [
      Object {
        "acf": Object {
          "featured_slider": false,
          "news_slider": true,
          "pinned": false,
          "should_push": false,
          "sponsored_by": "XYZ",
        },
        "categories": Array [
          6,
          1,
        ],
        "categories_names": Array [
          "Hotels",
          "News",
        ],
        "content": Object {
          "protected": false,
          "rendered": "<p>Durch smarte Raumanpassungen, große Flexibilität bei Buchungen und Stornierungen sowie die Etablierung eines extern zertifizierten und weit über den Standard hinaus gehenden Sicherheits- und Hygienekonzeptes hat sich Leonardo Hotels umfassend auf das „New Normal“ und die geänderten Anforderungen ihrer MICE-Kunden an zukunftsfähigen Veranstaltungsformaten eingestellt. Neue Pausenkonzepte, digitalen Helfer, verlässliche Partner sowie virtuelle und hybride Veranstaltungsformate umfassen die Angebotsvielfalt der Leonardo Hotels.  Diese sind überzeugt, dass persönliche Meetings mit Sicherheit wieder Zukunft haben.</p>
",
        },
        "date_gmt": "2020-11-11T08:25:41",
        "excerpt": Object {
          "protected": false,
          "rendered": "<p>Durch neue Sicherheits- und Hygienekonzeptes hat sich Leonardo Hotels umfassend auf das „New Normal“ und die geänderten Anforderungen an zukunftsfähigen Veranstaltungsformaten eingestellt.</p>
",
        },
        "featured_image_medium": "https://www.convention-net.de/wp-content/uploads/2020/11/iStock-1250329916-Personen_mit_Masken-w1500-h1300-600x400.jpg",
        "featured_image_thumb": "https://www.convention-net.de/wp-content/uploads/2020/11/iStock-1250329916-Personen_mit_Masken-w1500-h1300-300x300.jpg",
        "featured_media": 18571,
        "id": 18569,
        "title": Object {
          "rendered": "„New Normal“ bei Leonardo Hotels &#8211; mit neuen MICE-Formaten und Konzepten",
        },
      },
      Object {
        "acf": Object {
          "featured_slider": false,
          "news_slider": true,
          "pinned": false,
          "should_push": false,
          "sponsored_by": "",
        },
        "categories": Array [
          358,
          217,
          1,
        ],
        "categories_names": Array [
          "Branche",
          "Events",
          "News",
        ],
        "content": Object {
          "protected": false,
          "rendered": "<p><a href=\\"https://bit.ly/3iPkUhB\\">Cvent</a> CEO and Founder, Reggie Aggarwal, delivered an inspiring keynote to more than 7,000 event, marketing, and hospitality professionals at <a href=\\"https://bit.ly/30Unj4A\\">Cvent CONNECT Europe</a> today. In his opening presentation, Aggarwal urged attendees of the virtual event to seize the digital transformation opportunity in front of them to drive the industry’s recovery in 2021 and beyond.</p>
",
        },
        "date_gmt": "2020-11-10T11:00:19",
        "excerpt": Object {
          "protected": false,
          "rendered": "<p>Cvent CEO Reggie Aggarwal predicts “Golden Age of Events” during his keynote to more than 7,000 virtual attendees at Cvent CONNECT Europe </p>
",
        },
        "featured_image_medium": "",
        "featured_image_thumb": "",
        "featured_media": null,
        "id": 18592,
        "title": Object {
          "rendered": "Events and Hospitality Industries Urged to Harness the Power of Digital Transformation to Thrive in 2021 and Beyond",
        },
      },
    ]
  }
  decelerationRate="fast"
  directionalLockEnabled={true}
  disableVirtualization={false}
  enableMomentum={false}
  enableSnap={true}
  firstItem={0}
  forwardedRef={[Function]}
  getItem={[Function]}
  getItemCount={[Function]}
  hasParallaxImages={false}
  horizontal={true}
  inactiveSlideOpacity={0.5}
  inactiveSlideScale={0.8}
  inactiveSlideShift={0}
  initialNumToRender={6}
  invertStickyHeaders={false}
  inverted={false}
  itemWidth={750}
  keyExtractor={[Function]}
  layout="default"
  lockScrollTimeoutDuration={1000}
  lockScrollWhileSnapping={false}
  loop={false}
  loopClonesPerSide={3}
  maxToRenderPerBatch={13}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onResponderRelease={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  overScrollMode="never"
  pinchGestureEnabled={false}
  removeClippedSubviews={true}
  renderItem={[Function]}
  scrollEnabled={true}
  scrollEventThrottle={1}
  scrollsToTop={false}
  shouldOptimizeUpdates={true}
  showsHorizontalScrollIndicator={false}
  showsVerticalScrollIndicator={false}
  slideStyle={Object {}}
  sliderWidth={750}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "flexDirection": "row",
      "opacity": 0,
      "width": 750,
    }
  }
  swipeThreshold={20}
  updateCellsBatchingPeriod={50}
  useScrollView={false}
  vertical={false}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={13}
>
  <View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "opacity": 1,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": 750,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="featuredCarouselItem"
        >
          <View
            style={
              Object {
                "height": 400.2,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "flex": 1,
                  "resizeMode": "cover",
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://www.convention-net.de/wp-content/uploads/2020/11/iStock-1250329916-Personen_mit_Masken-w1500-h1300-300x300.jpg",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": undefined,
                      "width": undefined,
                    },
                    undefined,
                  ]
                }
              />
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "space-between",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignItems": "flex-end",
                      "backgroundColor": "rgba(223, 227, 232, 0.7)",
                      "color": "#333333",
                      "fontFamily": "sans",
                      "fontSize": 14,
                      "lineHeight": 17.5,
                      "paddingHorizontal": 30,
                      "paddingVertical": 10,
                      "textAlign": "right",
                      "width": "auto",
                    }
                  }
                >
                  Sponsored by 
                  <Text
                    style={
                      Object {
                        "color": "#333333",
                        "fontFamily": "sans-bold",
                        "fontSize": 15,
                        "lineHeight": 22.5,
                      }
                    }
                  >
                    XYZ
                  </Text>
                </Text>
                <View
                  style={
                    Object {
                      "backgroundColor": "rgba(223, 227, 232, 0.7)",
                      "justifyContent": "flex-end",
                      "paddingHorizontal": 30,
                      "paddingVertical": 10,
                    }
                  }
                >
                  <View
                    style={Object {}}
                  >
                    <Text
                      allowFontScaling={true}
                      selectable={false}
                      style={
                        Array [
                          Object {},
                          Object {},
                        ]
                      }
                    >
                      <Text
                        allowFontScaling={true}
                        style={
                          Object {
                            "fontFamily": "sans",
                            "fontSize": 14,
                            "lineHeight": 17.5,
                          }
                        }
                      >
                        „New Normal“ bei Leonardo Hotels – mit neuen MICE-Formaten und Konzepten
                      </Text>
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "opacity": 0.5,
            "transform": Array [
              Object {
                "scale": 0.8,
              },
            ],
            "width": 750,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="featuredCarouselItem"
        >
          <View
            style={
              Object {
                "height": 400.2,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "flex": 1,
                  "resizeMode": "cover",
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": undefined,
                      "width": undefined,
                    },
                    undefined,
                  ]
                }
              />
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "flex-end",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "rgba(223, 227, 232, 0.7)",
                      "justifyContent": "flex-end",
                      "paddingHorizontal": 30,
                      "paddingVertical": 10,
                    }
                  }
                >
                  <View
                    style={Object {}}
                  >
                    <Text
                      allowFontScaling={true}
                      selectable={false}
                      style={
                        Array [
                          Object {},
                          Object {},
                        ]
                      }
                    >
                      <Text
                        allowFontScaling={true}
                        style={
                          Object {
                            "fontFamily": "sans",
                            "fontSize": 14,
                            "lineHeight": 17.5,
                          }
                        }
                      >
                        Events and Hospitality Industries Urged to Harness the Power of Digital Transformation to Thrive in 2021 and Beyond
                      </Text>
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
